### PR TITLE
fix(kubernetes): support options.cascading in delete manifest pipelines

### PR DESCRIPTION
**Pull request by CalvinTse**
Previously, setting options.cascading in a delete manifest pipeline was a no-op. Now, it's another way (in addition to options.orphanDependents) to control whether kubernetes performs a cascading delete. If both
orphanDependents and cascading are specified, orphanDependents takes precedence. 
Also, this fix will allow clouddriver to look at the value of the "cascading" option in the Deck UI (which was not happening previously). 

The change was tested using unit tests with different pipelines as test cases (as seen in `clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/op/KubernetesDeleteManifestOperationTest.java`).
The change was also tested in a testing environment. In the testing environment, a pipeline with a Deploy and Delete Manifest Stage (with `{cascading: false}`) was created and executed, the cluster is then checked to see if its dependant objects have been deleted after a pipeline execution (If dependant objects still exist after the pipeline execution `{cascading: false}` then the change is working).

### CODE REVIEW

**Comment by dbyron-sf** [comment]
Mind splitting the commits into two -- one test() and one fix() so we can see the behavior change more clearly?

File: clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/op/KubernetesDeleteManifestOperationTest.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ void setupTest() {
   }
 
   @Test
-  public void orphanDependantsTrueTest() throws IOException {
+  public void orphanDependentsTrueTest() throws IOException {
```
Comment (dbyron-sf) [inline clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/op/KubernetesDeleteManifestOperationTest.java:99]: nit: ideally the initial commit wouldn't have the typos.

Comment (CalvinTse) [reply clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/op/KubernetesDeleteManifestOperationTest.java:99]: Fixed typo in first commit

Comment (dbyron-sf) [reply clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/op/KubernetesDeleteManifestOperationTest.java:99]: Thanks.

**Comment by dbyron-sf** [comment]
LGTM

**Review by dbyron-sf — APPROVED** [review]

**Review by clanesf — APPROVED** [review]
Looks good! Thank you for submitting this.

**Review by german-muzquiz — APPROVED** [review]

**Comment by german-muzquiz** [comment]
@Mergifyio backport release-1.26.x release-1.25.x

**Pull request merged**